src/textual/widgets/_list_view.py

```diff
@@ -52,6 +52,15 @@ def __init__(
         super().__init__(*children, name=name, id=id, classes=classes)
         self.index = initial_index
 
+    def on_mount(self) -> None:
+        """Ensure the ListView is fully-settled after mounting."""
+        # If someone inherits from ListView, they might be populating it
+        # from Widget.compose rather than from passing the ListItems to the
+        # initialisation call. Given that watch_index does all the work,
+        # let's just tickle it again after the DOM has been spun up and all
+        # children of this widget will have been mounted.
+        self.index = self.index or 0
```

Comment:
Small thing, but the validator would run twice I think.

I think we want to do something like `self._index = initial_index` inside the init, then in `on_mount` do `self.index = self._index`?

Reply:
I'll give that a shot.

Reply:
Done: 7544448002a6c382c5cd136933bbc7cf7465330d